recipes/default.rb

```diff
@@ -24,10 +24,21 @@
 # ::Chef::Recipe.send(:include, Chef::Mixin::PowershellOut)
 ::Chef::Resource::RubyBlock.send(:include, Chef::Mixin::PowershellOut)
 
-powershell_script 'install chocolatey' do
-  code "iex ((new-object net.webclient).DownloadString('#{node['chocolatey']['Uri']}'))"
-  convert_boolean_return true
-  not_if { ChocolateyHelpers.chocolatey_installed? }
+if File.exist?('C:\windows\system32\cmd.exe')
+  arch = :x86_64
+  cmd = 'C:\windows\system32\cmd.exe'
```

Comment:
On a 32 bit system the path `C:\windows\system32\cmd.exe` is going to exist, which means this will always set arch to `:x86_64`.

I would use something more deterministic for setting the architecture.

On the whole, you should look for the existence of `c:\Windows\sysnative\cmd.exe`, which is going to tell you that you have a 32 bit process on a 64 bit system and SysWOW64 is in play.